contracts/Zauction.sol

```diff
@@ -121,6 +139,56 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     );
   }
 
+  function setBuyPrice(uint256 amount, uint256 tokenId) external {
+    address owner = registrar.ownerOf(tokenId);
+    require(msg.sender == owner, "zAuction: only owner can set price");
+    priceInfo[tokenId] = Listing(amount, owner);
+  }
+
+  /// recovers buyer's signature based on seller's proposed data and, if bid data hash matches the message hash, transfers nft and payment
+  /// @param amount token amount of sale
+  /// @param tokenId token id we are transferring
+  /// @param startBlock block number at which acceptBid starts working
+  /// @param expireBlock block number at which acceptBid stops working
+  function buyNow(
+    uint256 amount,
+    uint256 tokenId,
+    uint256 startBlock,
+    uint256 expireBlock
+  ) external {
+    require(startBlock <= block.number, "zAuction: sale hasn't started");
```

Comment:
`startBlock` and `expireBlock` are both parameters here, why would I as a user ever purposely set `startBlock` past the current block number and then submit the transaction? It would be an automatic transaction failure.

What purpose does `startBlock` and `expireBlock` have in this code? Is it a requirement for something, does it protect against something?

Reply:
There might be a reason for `expireBlock` to exist here though (can help prevent front-running attacks potentially, or other attacks like this) we can talk more about this if you want, but I want you to have a clear explanation / understanding on why each line of code is there and what it's doing at a meta-perspective.

Keep in mind, every operation = more gas to run = more expensive. We want to reduce the cost of interacting with our smart contracts, but not at the cost of security/saftey.